lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -732,6 +732,306 @@ class DecomposeAtenTriuOp : public OpRewritePattern<AtenTriuOp> {
+                                       /*dtype=*/dtype, /*layout=*/layout,
+                                       /*device=*/device,
+                                       /*pin_memory=*/pin_memory);
+}
+
+void _getTrilSizes(PatternRewriter &rewriter, Location loc, int64_t row,
+                   int64_t col, int64_t offset,
+                   /*return values*/ int64_t &trapezoidSize,
+                   int64_t &rectangleSize) {
+
+  // Base case
+  if (row == 0 || col == 0) {
+    trapezoidSize = 0;
+    rectangleSize = 0;
+
+    return;
+  }
+
+  // Calculate mFirstRow size
+  int64_t mFirstRow;
+  if (offset > 0)
+    mFirstRow = (col < offset + 1) ? col : offset + 1;
+  else
+    mFirstRow = (row + offset > 0) ? 1 : 0;
+
+  // Calculate mLastRow size
+  int64_t minimum = (col < row + offset) ? col : row + offset;
+  int64_t mLastRow = (minimum > 0) ? minimum : 0;
+
+  // Calculate nRowAll
+  minimum = (row < row + offset) ? row : row + offset;
+  int64_t nRowAll = (minimum > 0) ? minimum : 0;
+
+  // Calucltae nRowTrapezoid
+  int64_t nRowTrapezoid = mLastRow - mFirstRow + 1;
+
+  // Number of elements in top trapezoid - trapezoidSize
+  trapezoidSize = (mFirstRow + mLastRow) * nRowTrapezoid / 2;
+
+  // Number of elements in bottom rectangle - rectangleSize
+  int64_t diffRow = nRowAll - nRowTrapezoid;
+  rectangleSize = (diffRow * col > 0) ? diffRow * col : 0;
+}
+
+void _getTriuSizes(PatternRewriter &rewriter, Location loc, int64_t row,
+                   int64_t col, int64_t offset,
+                   /*return values*/ Value &trapezoidSize, Value &rectangleSize,
+                   Value &mFirstRow) {
+
+  // Constants
+  Value cstZero = rewriter.create<Torch::ConstantIntOp>(loc, 0);
+
+  // Base case
+  if (row == 0 || col == 0) {
+    trapezoidSize = cstZero;
+    rectangleSize = cstZero;
+    mFirstRow = cstZero;
+
+    return;
+  }
+
+  // Calculate mFirstRow size
+  int64_t maximum = (col - offset > 0) ? col - offset : 0;
+  int64_t mFirstRowInt = (offset > 0) ? maximum : col;
+
+  // Number of elements in top rectangle - calculate rectangle size
+  int64_t minimum = (row < -offset) ? row : -offset;
+  int64_t rectangleSizeInt = (minimum * col > 0) ? minimum * col : 0;
+
+  // Number of elements in bottom trapezoid - calculte trapezoid size
+  int64_t trapezoidSizeTril;
+  int64_t rectangleSizeTril;
+
+  _getTrilSizes(rewriter, loc, row, col, offset - 1, trapezoidSizeTril,
+                rectangleSizeTril);
+  int64_t triuSize = row * col - (trapezoidSizeTril + rectangleSizeTril);
+  int64_t trapezoidSizeInt = triuSize - rectangleSizeInt;
+
+  // Create Value from int
+  trapezoidSize = rewriter.create<Torch::ConstantIntOp>(
+      loc, rewriter.getI64IntegerAttr(trapezoidSizeInt));
+  rectangleSize = rewriter.create<Torch::ConstantIntOp>(
+      loc, rewriter.getI64IntegerAttr(rectangleSizeInt));
+  mFirstRow = rewriter.create<Torch::ConstantIntOp>(
+      loc, rewriter.getI64IntegerAttr(mFirstRowInt));
```

Comment:
nit: I would remove this wrapping in MLIR `Value`s to outside the function for two reasons:

1. The current use of this helper is followed by a `matchPattern` to unwrap the `Value`s, so it makes sense to keep the values as C++ `int`s until they are no longer needed in such a form
2. Returning just C++ `int`s means that both helpers (`getTrilSizes` and `getTriuSizes`) have consistent signatures